src/containers/community/CommunityJoin.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { connect } from 'react-redux'
+const { func, bool } = React.PropTypes
+import { get, isEmpty } from 'lodash'
+import { joinCommunityWithCode, navigate, resetCommunityValidation, validateCommunityAttribute } from '../../actions'
+import cx from 'classnames'
+
+@connect(({communityValidation}) => ({
+  codeInvalid: get(communityValidation, 'beta_access_code.unique')
+}))
+export default class CommunityJoin extends React.Component {
+  static propTypes = {
+    dispatch: func,
+    codeInvalid: bool
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {codeEmpty: true}
+  }
+
+  codeChanged = event => {
+    let { dispatch } = this.props
+    let { value } = event.target
+    if (isEmpty(value.trim())) {
+      this.setState({codeEmpty: true})
```

Comment:
i think state is fine. you could even skip `setState` entirely and just check `this.refs.code.value` in `render()`